crates/gateway-client/src/lib.rs

```diff
@@ -446,6 +458,72 @@ impl ClientApi for Client {
     }
 }
 
+#[cfg(any(test, feature = "test-utils"))]
+pub mod utils {
+    use anyhow::Context;
+    use pathfinder_common::{BlockId, Chain};
+    use semver::Version;
+    use tokio::sync::OnceCell;
+
+    use crate::{Client, ClientApi};
+
+    /// Returns true if the latest block's version matches the requirement.
+    /// 
+    /// This is useful to add version checks for deprecation items or changes that
+    /// should be made when a certain version reaches a specifc network.
+    /// 
+    /// If the gateway query fails, then this still returns true - this is to prevent
+    /// failures due to rate limiting or if the gateway is down. In other words this
+    /// may give false positives if the gateway is unavailable for whatever reason.
+    pub async fn gateway_version_check(network: Chain, version_requirement: &str) -> bool {
+        static INTEGRATION: OnceCell<Version> = OnceCell::const_new();
+        static TESTNET: OnceCell<Version> = OnceCell::const_new();
+        static TESTNET2: OnceCell<Version> = OnceCell::const_new();
+        static MAINNET: OnceCell<Version> = OnceCell::const_new();
+
+        let req = semver::VersionReq::parse(version_requirement).unwrap();
+
+        let version = match network {
+            Chain::Mainnet => MAINNET
+                .get_or_try_init(|| async { get_latest_version(&Client::mainnet()).await })
+                .await
+                .cloned(),
+            Chain::Testnet => TESTNET
+                .get_or_try_init(|| async { get_latest_version(&Client::testnet()).await })
+                .await
+                .cloned(),
+            Chain::Testnet2 => TESTNET2
+                .get_or_try_init(|| async { get_latest_version(&Client::testnet2()).await })
+                .await
+                .cloned(),
+            Chain::Integration => INTEGRATION
+                .get_or_try_init(|| async { get_latest_version(&Client::integration()).await })
+                .await
+                .cloned(),
+            _ => panic!("Custom networks are not supported"),
+        };
+
+        let version = match version {
+            Ok(v) => v,
+            Err(_) => return true,
```

Comment:
This seems like a non-strict check. Why not propagate the error and let client code decide?